ush/forecast_postdet.sh

```diff
@@ -497,229 +469,198 @@ EOF
+    while [[ ${idate} -le ${forecast_end_cycle} ]]; do
+      for file in "${idate:0:8}.${idate:8:2}0000."*; do
+        ${NCP} "${file}" "${COM_ATMOS_RESTART}/${file}"
       done
-      if [ $DOIAU = "YES" ] || [ $DOIAU_coldstart = "YES" ]; then
-        # if IAU is on, save restart at start of IAU window
-        rst_iau=$(( ${IAU_OFFSET} - (${IAU_DELTHRS}/2) ))
-        if [ $rst_iau -lt 0 ];then
-          rst_iau=$(( (${IAU_DELTHRS}) - ${IAU_OFFSET} ))
-        fi
-        RDATE=$($NDATE +$rst_iau $CDATE)
-        rPDY=$(echo $RDATE | cut -c1-8)
-        rcyc=$(echo $RDATE | cut -c9-10)
-        for file in "${rPDY}.${rcyc}0000."* ; do
-          ${NCP} "${file}" "${COM_ATMOS_RESTART}/${file}"
-        done
-      fi
-    elif [[ ${RUN} =~ "gfs" ]]; then
-      ${NCP} "${DATA}/input.nml" "${COM_ATMOS_HISTORY}/input.nml"
-    fi
+      local idate=$(date -d "${idate:0:8} ${idate:8:2} + ${restart_interval} hours" +%Y%m%d%H)
+    done
+  elif [[ ${RUN} =~ "gfs" ]]; then
+    # No need to copy FV3 restart files when RUN=gfs
+    ${NCP} "${DATA}/input.nml" "${COM_ATMOS_HISTORY}/input.nml"
   fi
-
   echo "SUB ${FUNCNAME[0]}: Output data for FV3 copied"
 }
 
-
 WW3_postdet() {
   echo "SUB ${FUNCNAME[0]}: Linking input data for WW3"
   COMPONENTwave=${COMPONENTwave:-${RUN}wave}
 
   #Link mod_def files for wave grids
-  if [ $waveMULTIGRID = ".true." ]; then
-    array=($WAVECUR_FID $WAVEICE_FID $WAVEWND_FID $waveuoutpGRD $waveGRD $waveesmfGRD)
-    echo "Wave Grids: $WAVECUR_FID $WAVEICE_FID $WAVEWND_FID $waveuoutpGRD $waveGRD $waveesmfGRD"
+  if [[ ${waveMULTIGRID} = ".true." ]]; then
+    array=(${WAVECUR_FID} ${WAVEICE_FID} ${WAVEWND_FID} ${waveuoutpGRD} ${waveGRD} ${waveesmfGRD})
+    echo "Wave Grids: ${WAVECUR_FID} ${WAVEICE_FID} ${WAVEWND_FID} ${waveuoutpGRD} ${waveGRD} ${waveesmfGRD}"
     grdALL=$(printf "%s\n" "${array[@]}" | sort -u | tr '\n' ' ')
 
     for wavGRD in ${grdALL}; do
       ${NCP} "${COM_WAVE_PREP}/${COMPONENTwave}.mod_def.${wavGRD}" "${DATA}/mod_def.${wavGRD}"
     done
-  else 
-    #if shel, only 1 waveGRD which is linked to mod_def.ww3 
+  else
+    #if shel, only 1 waveGRD which is linked to mod_def.ww3
     ${NCP} "${COM_WAVE_PREP}/${COMPONENTwave}.mod_def.${waveGRD}" "${DATA}/mod_def.ww3"
   fi
 
 
   #if wave mesh is not the same as the ocn/ice mesh, linkk it in the file
   comparemesh=${MESH_OCN_ICE:-"mesh.mx${ICERES}.nc"}
-  if [ "$MESH_WAV" = "$comparemesh" ]; then
+  if [[ "${MESH_WAV}" = "${comparemesh}" ]]; then
     echo "Wave is on same mesh as ocean/ice"
   else
-    $NLN -sf $FIXwave/$MESH_WAV $DATA/
+    ${NLN} -sf "${FIXwave}"/"${MESH_WAV}" "${DATA}"/
   fi
 
   export wavprfx=${RUNwave}${WAV_MEMBER:-}
 
   #Copy initial condition files:
-  for wavGRD in $waveGRD ; do
-    if [ $warm_start = ".true." -o $RERUN = "YES" ]; then
-      if [ $RERUN = "NO" ]; then
+  for wavGRD in ${waveGRD} ; do
+    if [ "${warm_start}" = ".true." -o "${RERUN}" = "YES" ]; then
```

Comment:
Shellcheck seems to have this weird tic where if you use `-a` or `-o` in single brackets, it doesn't try to have you change either.
```suggestion
    if [[ "${warm_start}" = ".true." ]] || [[ "${RERUN}" = "YES" ]]; then
```